test/access/manager/AuthorityUtils.test.js

```diff
@@ -0,0 +1,91 @@
+require('@openzeppelin/test-helpers');
+
+const AuthorityUtils = artifacts.require('$AuthorityUtils');
+const NotAuthorityMock = artifacts.require('NotAuthorityMock');
+const AuthorityNoDelayMock = artifacts.require('AuthorityNoDelayMock');
+const AuthorityDelayMock = artifacts.require('AuthorityDelayMock');
+const AuthorityNoResponse = artifacts.require('AuthorityNoResponse');
+
+contract('AuthorityUtils', function (accounts) {
+  const [user, other] = accounts;
+
+  beforeEach(async function () {
+    this.mock = await AuthorityUtils.new();
+  });
+
+  describe('canCallWithDelay', function () {
+    describe('when authority does not have a canCall function', function () {
+      beforeEach(async function () {
+        this.authority = await NotAuthorityMock.new();
+      });
+
+      it('returns (immediate = 0, delay = 0)', async function () {
+        const { immediate, delay } = await this.mock.$canCallWithDelay(
+          this.authority.address,
+          user,
+          other,
+          '0x12345678',
+        );
+        expect(immediate).to.equal(false);
+        expect(delay).to.be.bignumber.equal('0');
+      });
+    });
+
+    describe('when authority has no delay', function () {
+      beforeEach(async function () {
+        this.authority = await AuthorityNoDelayMock.new();
+        this.immediate = true;
+        await this.authority._setImmediate(this.immediate);
+      });
+
+      it('returns (immediate, delay = 0)', async function () {
+        const { immediate, delay } = await this.mock.$canCallWithDelay(
+          this.authority.address,
+          user,
+          other,
+          '0x12345678',
+        );
+        expect(immediate).to.equal(this.immediate);
+        expect(delay).to.be.bignumber.equal('0');
+      });
+    });
+
+    describe('when authority replies with a delay', function () {
+      beforeEach(async function () {
+        this.authority = await AuthorityDelayMock.new();
+        this.immediate = true;
+        this.delay = web3.utils.toBN(42);
+        await this.authority._setImmediate(this.immediate);
+        await this.authority._setDelay(this.delay);
+      });
+
+      it('returns (immediate, delay)', async function () {
+        const { immediate, delay } = await this.mock.$canCallWithDelay(
+          this.authority.address,
+          user,
+          other,
+          '0x12345678',
+        );
+        expect(immediate).to.equal(this.immediate);
+        expect(delay).to.be.bignumber.equal(this.delay);
+      });
+    });
```

Comment:
This is actually not something we expect to see. Its ok to test it, but then we should also test `false`, `delay`

Maybe we should have two for loops:
```
for (const immediate of [ true, false ]) {
  for (const delay of [ 0, 42 ].map(web3.utils.toBN)) {
    it(`returns (immediate=${immediate}, delay=${delay})`, async function () {
      const authority = await AuthorityDelayMock.new();
      await authority._setImmediate(this.immediate);
      await authority._setDelay(delay);
      const result = await this.mock.$canCallWithDelay(
          this.authority.address,
          user,
          other,
          '0x12345678',
        );
        expect(result.immediate).to.equal(immediate);
        expect(result.delay).to.be.bignumber.equal(delay);
    }
  }
}
```